### PR TITLE
combined commit of two new datareducers and method for DROs to write …

### PR DESCRIPTION
**Pull request by markusbattarbee**
…parameters, tested with master. This is so we get these also in Vlasiator-4.

### CODE REVIEW

**Comment by ursg** [comment]
Just for completeness: This contains the contents of pull requests https://github.com/fmihpc/vlasiator/pull/403 and https://github.com/fmihpc/vlasiator/pull/411, except they go straight into the master branch instead of dev.

**Comment by galfthan** [comment]
The commits are different even if code is the same, does it cause merge conflicts?

**Comment by ursg** [comment]
Not if they are cherry-picks, but this is a squashed combination of them, right?

**Comment by galfthan** [comment]
Ok, the merge is fine for me .

File: parameters.cpp [hunk 1]
```diff
@@ -210,7 +210,7 @@ bool Parameters::addParameters(){
...
+   Readparameters::addComposing("variables.output", "List of data reduction operators (DROs) to add to the grid file output. Each variable to be added has to be on a new line output = XXX. Available (20190320): B BackgroundB PerturbedB E Rhom Rhoq populations_Rho V populations_V populations_moments_Backstream populations_moments_NonBackstream populations_EffectiveSparsityThreshold populations_EnergyDensity populations_PrecipitationFlux populations_RhoLossAdjust LBweight MaxVdt MaxRdt populations_MaxVdt populations_MaxRdt MaxFieldsdt MPIrank FsGridRank FsGridBoundaryType BoundaryType BoundaryLayer populations_Blocks fSaved populations_accSubcycles VolE HallE GradPeE VolB BackgroundVolB PerturbedVolB Pressure populations_PTensor derivs BVOLderivs GridCoordinates Potential BackgroundVolE ChargeDensity PotentialError MeshData");
```
Comment (ykempf) [inline parameters.cpp]: Was this not supposed to be in the broken up format?

Comment (markusbattarbee) [reply parameters.cpp]: in dev, not in master

Comment (ykempf) [reply parameters.cpp]: OK, worst case it'll be an easy conflict to resolve.

**Comment by ykempf** [comment]
Please disregard my self-assigned request for review, now I know what that one button was for.

**Comment by ursg** [comment]
Doesn't break anything and runs the testpackage with zero diffs. Good to go!

**Pull request merged**